data-prepper-plugins/mutate-string-processors/README.md

```diff
@@ -0,0 +1,106 @@
+# Mutate String Processors
+The following is a list of processors to mutate a string.
+
+## SubstituteStringProcessor
+A processor that matches a key's value against a regular expression and replaces all matches with a replacement string.
+
+### Example
+The following `substitute_string` processor example will replace all ':' with '-' for the `message` key.
```

Comment:
I think it would be good to have a sample input. You did something similar with `SplitStringProcessor` with `"hello,world"` that I think is good.

Reply:
I think it makes sense to have the `Basic Usage` section instead since I need the `yaml` processor config and also the `logs_json.log` for users to understand that's where they place the values of Strings to be manipulated.

Reply:
I just mean to say something like the following. You can word it differently if you prefer.

> The following `substitute_string` processor example will replace all ':' with '-' for the `message` key. The processor will change a `message` value of `"hello:world"` to `"hello-world"`.

Reply:
I just included a `Basic Usage` section which has a sample `pipeline.yaml` and `logs_json.log` to make it clear to the user what needs to be updated and quickly run the `pipeline.yaml` with Data Prepper and see it work.